framework.py

```diff
@@ -9,16 +9,150 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+from sgtk.util import filesystem
+
+import threading
+import datetime
+import os
+import time
 
 class ShotgunUtilsFramework(sgtk.platform.Framework):
     
+    # A list of file names which should never be deleted when cleaning up old
+    # cached data.
+    _ALWAYS_KEEP_CACHED_FILES = [
+        "sg_schema.pickle",
+        "sg_status.pickle",
+    ]
     ##########################################################################################
     # init and destroy
             
     def init_framework(self):
+        """
+        Init this framework.
+
+        Post an old cached data cleanup in the background
+        """
         self.log_debug("%s: Initializing..." % self)
+        self._stop_cleanup = False
+        self._bg_cleanup_thread = None
+        self._post_old_data_cleanup()
     
     def destroy_framework(self):
+        """
+        Destroy this framework.
+
+        If an old cached data cleanup was posted in the background, stop it
+        immediately.
+        """
         self.log_debug("%s: Destroying..." % self)
+        # Please note that we are modifying a member which is read in another
+        # thread which should be fine in Python with the GIL protecting its access.
+        self._stop_cleanup = True
+        if self._bg_cleanup_thread:
+            if self._bg_cleanup_thread.isAlive():
+                # If the clean up is not completed yet, log why we are waiting.
+                self.log_info("Waiting for old data clean up to complete...")
+            self._bg_cleanup_thread.join()
     
-    
+    def _post_old_data_cleanup(self):
+        """
+        Post a cleanup of old cached data in the background.
+        """
+        try:
+            self.log_debug(
+                "Posting old cached data clean up..."
+            )
+            self._stop_cleanup = False
+            # Qt might not be yet available at this stage (e.g. in tk-desktop),
+            # so we can't use a background task manager or a QThread, use instead
+            # regular Python Thread to post the clean up in the background.
+            self._bg_cleanup_thread = threading.Thread(
+                target=self._remove_old_cached_data,
+                name="%s Clean Up" % self.name
+            )
+            self._bg_cleanup_thread.start()
+        except Exception as e:
+            self.log_warning("Unable to post data clean up: %s" % e)
+
+    def _remove_old_cached_data(self, grace_period=7):
```

Comment:
Would be good if we defined the grace_period as a big constant that belongs to the class. And set it to be 60 days.